srml/aura/src/lib.rs

```diff
@@ -93,59 +114,66 @@ impl AuraReport {
 	}
 }
 
-impl<T: Trait> Module<T>
-	where T::OnOfflineValidator: consensus::OnOfflineValidator<AuraReport>,
-{
-	fn on_timestamp_set(now: T::Moment, step_duration: T::Moment) {
+impl<T: Trait> Module<T> {
+	/// Determine the Aura slot-duration based on the timestamp module configuration.
+	pub fn slot_duration() -> u64 {
+		// we double the minimum block-period so each author can always propose within
+		// the majority of their slot.
+		<timestamp::Module<T>>::block_period().as_().saturating_mul(2)
+	}
+
+	/// Verify an inherent slot that is used in a block seal against a timestamp
+	/// extracted from the block.
+	// TODO: ensure `ProvideInherent` can deal with dependencies like this.
```

Comment:
Could you please open an issue for this?

Reply:
Done